[DM-3601] device operations goes directly to pending state (#2005)
SoftwareAG/c8y-docs: content/change-logs/device-management/cumulocity-10-18-540-112-device-operations-goes-directly-to-pending.md

---
date: 2024-04-10T15:05:45.132Z

> title: LWM2M devices operations goes directly to Pending state

Reviewer: ```suggestion
title: LWM2M device operations directly go to Pending state
```

product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-1KLUzmqfe
    label: LWM2M
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: DM-3601
version: 10.18.540.112
---

> When the LWM2M device sends LWM2M registration or registration update to the LWM2M agent, the LWM2M agent also subscribes to the realtime channel so that user defined operations are immediately executed on this device during the instance level time to live configuration period of the LWM2M agent.

Reviewer: ```suggestion
When the LWM2M device sends a registration or registration update to the LWM2M agent, the LWM2M agent also subscribes to the real-time channel so that user-defined operations are immediately executed on this device during the instance level time-to-live configuration period of the LWM2M agent.
```

> When the LWM2M device sends LWM2M registration or registration update to the LWM2M agent, the LWM2M agent also subscribes to the realtime channel so that user defined operations are immediately executed on this device during the instance level time to live configuration period of the LWM2M agent.

Reviewer 2: Elias here is like:
Introduction on the area the change is relevant to.
Two changes that are done.

To make user grasp the idea easier.

The changes are added for the following cases:

> Firstly, when the device does a registration update to the LWM2M server within the previous realtime subscription period, then the subscription expiry period was not extended and the device is automatically unsubscribed from realtime.

Reviewer: ```suggestion
Firstly, when the device does a registration update to the LWM2M server within the previous real-time subscription period, then the subscription expiry period was not extended and the device was automatically unsubscribed from realtime.
```

> This has been resolved by renewing the realtime subscription period every time when the device does a registration update.

Reviewer: ```suggestion
This has been resolved by renewing the real-time subscription period every time when the device does a registration update.
```

> Secondly, when a device operation is created, this operation extended the realtime subscription based on the instance level time to live extension period defined for the LWM2M agent for the cases of the realtime time to live period being expired earlier than this extension.

Reviewer: ```suggestion
Secondly, when a device operation is created, this operation extended the real-time subscription based on the instance level time-to-live extension period defined for the LWM2M agent for the cases of the real-time time-to-live period being expired earlier than this extension.
```

> Secondly, when a device operation is created, this operation extended the real-time subscription based on the instance level time-to-live extension period defined for the LWM2M agent for the cases of the real-time time-to-live period being expired earlier than this extension.

Reviewer: This sentence is hard to read. Any idea how to rephrase it without losing the meaning?

> This is now changed to add the extension on top of realtime subscription time to live period in order to have longer realtime subscription periods.

Reviewer: ```suggestion
This is now changed to add the extension on top of the real-time subscription time-to-live period in order to have longer real-time subscription periods.
```
